build-support/docker/centos6/Dockerfile

```
@@ -4,18 +4,33 @@
 # Use centos6 for compatibility with glibc 2.12.
 FROM centos:6
 
-# Install python 2.7.13, (more) modern gcc, and a JDK.
+# Install a (more) modern gcc, and a JDK.
+# Also install sqlite-devel because python 3 errors on startup if not -- see
+# https://stackoverflow.com/a/39907500/2518889.
 RUN yum -y update
+# TODO: figure out why this needs to be installed first for /usr/bin/scl to work!
 RUN yum install -y centos-release-scl
-RUN yum install -y \
-  devtoolset-7-gcc \
-  devtoolset-7-gcc-c++ \
+RUN yum install -y devtoolset-7-gcc{,-c++} \
   git \
   java-1.8.0-openjdk-devel \
-  python27 \
-  libffi \
-  libffi-devel \
-  openssl-devel
+  libffi{,-devel} \
+  openssl-devel \
+  sqlite-devel
+
+# Install the desired versions of python 2 and 3.
+ARG PYTHON_2_VERSION=2.7.13
+ARG PYTHON_3_VERSION=3.6.8
+# NB: PYENV_ROOT must be set in the environment for `pyenv install` to work, otherwise it will
+# mysteriously fail to find the `install` command.
+ENV PYENV_ROOT /pyenv-docker-build
+RUN mkdir ${PYENV_ROOT}
+RUN git clone https://github.com/pyenv/pyenv ${PYENV_ROOT}
+# Install python 2 and 3, and set python 2 as the default interpreter.
+RUN /usr/bin/scl enable devtoolset-7 -- bash -c '\
+    ${PYENV_ROOT}/bin/pyenv install --keep --skip-existing ${PYTHON_2_VERSION} \
+    && ${PYENV_ROOT}/bin/pyenv install --keep --skip-existing ${PYTHON_3_VERSION} \
+    && ${PYENV_ROOT}/bin/pyenv global ${PYTHON_2_VERSION}'
```

We need to set both Py2 and Py3 as available through `pyenv global`, otherwise we get this issue https://github.com/pyenv/pyenv/issues/34. The bins we then resolve as we'd expect, `python` -> 2.7, `python2` -> 2.7, and `python3` -> 3.6.

```suggestion
    && ${PYENV_ROOT}/bin/pyenv global ${PYTHON_2_VERSION} ${PYTHON_3_VERSION}' 
```

TIL that `pyenv global` allows multiple arguments..

---

Well, I wouldn't expect it to work without that line unless you first run `pyenv` to set up your PATH... I'm just arguing that we should run `pyenv` commands for that instead of directly manipulating PATH, so that all pyenv-related PATH manipulations are done "through the front door" so to speak, to avoid confusion.

---

See https://github.com/pantsbuild/pants/pull/7064#discussion_r247742206 -- no clue why `pyenv-installer` fails to do its job here, it may be because we are running as root in the Docker container and it doesn't expect that, but regardless we now use the pyenv initialization commands in a `.bashrc`.